### PR TITLE
Components

### PR DESCRIPTION
**Pull request by erikpukinskis**
This patch exports a `component` method that lets you build your own renderables with a similar invocation style to teacup tags and can be easily mixed with them.

Examples in test/component.coffee should be pretty self explanatory!

### CODE REVIEW

**Comment by hurrymaplelad** [comment]
Awesome!  Thanks for putting this together.

Couple thoughts after talking through it w/ @erikpukinskis:

We could support `closableModal` style content functions (ex `({closeButton}) ->`) by extending `renderContents` signature to take `renderContents: (contents, rest...) ->` then passing `rest` through if contents is a function: `contents.apply @, rest`.

Would love to see example usage in the README.

Garbanzo form_helpers could be a cool place to establish this convention in our code base.  Most (all?) of the field helpers could be simplified using the new `component` method.

**Comment by hurrymaplelad** [comment]
Merge at will!

File: test/component.coffee [hunk 1]
```diff
@@ -0,0 +1,46 @@
+expect = require 'expect.js'
+{component, div, render, img, text, button} = require '..'
+
+console.log {component, div, render}
```
Comment (asalant) [inline test/component.coffee]: Intended?

Comment (erikpukinskis) [inline test/component.coffee]: I noticed it after I merged! It's out of master.

**Pull request merged**